pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -386,14 +398,16 @@ func (am *Alertmanager) applyConfig(cfg *apimodels.PostableUserConfig, rawConfig
 	}
 
 	am.inhibitor = inhibit.NewInhibitor(am.alerts, cfg.AlertmanagerConfig.InhibitRules, am.marker, am.gokitLogger)
+	am.muteTimes = am.buildMuteTimesMap(cfg.AlertmanagerConfig.MuteTimeIntervals)
 	am.silencer = silence.NewSilencer(am.silences, am.marker, am.gokitLogger)
 
 	meshStage := notify.NewGossipSettleStage(am.peer)
 	inhibitionStage := notify.NewMuteStage(am.inhibitor)
+	timeMuteStage := notify.NewTimeMuteStage(am.muteTimes)
 	silencingStage := notify.NewMuteStage(am.silencer)
 	for name := range integrationsMap {
 		stage := am.createReceiverStage(name, integrationsMap[name], am.waitFunc, am.notificationLog)
-		routingStage[name] = notify.MultiStage{meshStage, silencingStage, inhibitionStage, stage}
+		routingStage[name] = notify.MultiStage{meshStage, silencingStage, timeMuteStage, inhibitionStage, stage}
```

Comment:
Should `timeMuteStage` go before `silencingStage` or this the best place for it? I'm not sure which is best, perhaps there is a good justification for putting it after `silencingStage`?

Reply:
I think the order does not matter, as those stages act as a filter before the last stage, which is the receiver stage where the notification will happen.